### PR TITLE
feat(menu): add optional danger state for menu items

### LINKED ISSUE #8058: Menu- add optional danger state for menu items
Want to allow for applying "danger' styling to menu items.

Core PR: https://github.com/patternfly/patternfly/pull/5102

### PR DESCRIPTION
**Pull request by gefgu**
**What**: Closes #8058

Because most of the style in `MenuItem.tsx` is applied based on a boolean prop, I've decided to do the same here.

### CODE REVIEW

**Comment by gefgu** [comment]
This is a new feature. In this way, should I add an example in [docs](https://github.com/patternfly/patternfly-react/blob/main/packages/react-core/src/components/Menu/examples/Menu.md) showcasing it?

**Review by jenny-s51 — CHANGES_REQUESTED** [review]
New prop looks great @gefgu ! 

Regarding the demo, yes please - here is what the demo should look like based on the HTML example in core: https://patternfly-pr-5102.surge.sh/components/menu#danger-menu-item 

[Related PR for reference
](https://github.com/patternfly/patternfly/pull/5102)

**Comment by gefgu** [comment]
@jenny-s51 Thanks for the feedback!

I've built the demo and added it to the end of the list because of [this comment](https://github.com/patternfly/patternfly/pull/5102#pullrequestreview-1116199160)

File: packages/react-core/src/components/Menu/examples/MenuDangerMenuItem.tsx
Comment (gefgu) [inline packages/react-core/src/components/Menu/examples/MenuDangerMenuItem.tsx:10]: I've made this log call for consistency with the other examples.

**Review by jenny-s51 — APPROVED** [review]
> @jenny-s51 Thanks for the feedback!
> 
> I've built the demo and added it to the end of the list because of [this comment](https://github.com/patternfly/patternfly/pull/5102#pullrequestreview-1116199160)

LGTM @gefgu , thank you for your work on this! Adding the demo to the end makes sense given @mcoker's comment in the core issue 👍

File: packages/react-core/src/components/Menu/MenuItem.tsx
Comment (mcoker) [inline packages/react-core/src/components/Menu/MenuItem.tsx:301]: Typically for these statuses, we include some sr-only text https://github.com/patternfly/patternfly-react/blob/60f9853c6617adbc51dfd015f03a731c5141b421/packages/react-core/src/components/Popover/Popover.tsx#L452

We have the sr-only text in the core example. Should we add that? cc @thatblindgeye

Comment (thatblindgeye) [reply packages/react-core/src/components/Menu/MenuItem.tsx:301]: This feels like an interesting scenario to me. I think it could really go either way, but it would bring into question whether the "danger"/"warning" variant of our Button component should have similar visually hidden text.

I don't believe I've seen visually hidden text like this used for "danger" menu items/buttons before, and it could possibly stray into being too verbose (especially the more visually hidden text there is on a page)... but at the same time, if we want to visually convey these menu items as "danger" items to users that are sighted, regardless if there's an additional step from a confirmation modal, it could make sense to convey that a menu item is a "danger" item to users of AT similarly.

Looking at Web-AIM's [invisible content](https://webaim.org/techniques/css/invisiblecontent/) page in the "Examples" section, they have a note about:

>Use these techniques judiciously! Keep in mind that many screen reader users have some vision—what they see and what they hear should typically be in harmony. In general, screen reader-only content should be reserved for information is apparent visually but not apparent to blind screen reader users.

Which this scenario could technically fit into that sort of criteria (a "danger" item is apparent visually via the text color, but not as such to users that are blind and navigating via SR).

It could possibly come down to whether the text of the "danger" menu item is enough to convey that it is a dangerous/destructive action (i.e. will these dangerous items always be as clear as "Delete"), or whether the additional context via color and/or visually hidden text is needed to help convey that dangerous/destruction nature (i.e. is the text color more to convey an item as dangerous/destructive, or more to just visually separate it from other menu items?).

I think I'd generally be okay with SR text being added, but it would probably lead to a broader convo regarding other components like Button that also have status/severity sort of visual styling.

**Review by mmenestr — APPROVED** [review]
LGTM!

File: packages/react-core/src/components/Menu/examples/Menu.md
Comment (mcoker) [reply packages/react-core/src/components/Menu/examples/Menu.md:122]: @tlabaj oh, I made a comment in the core PR to move it after the "view more" and "loading" examples, since those seem like similar variations, but I missed that it wasn't made before merging. WDYT? https://github.com/patternfly/patternfly/pull/5102#pullrequestreview-1116199160

**Review by mcoker — APPROVED** [review]
LGTM! If you end up moving the example, I'll create a follow up issue in core.

**Comment by gefgu** [comment]
@tlabaj I've moved the example up so it is second in page and matches the html examples.

**Review by tlabaj — APPROVED** [review]
LGTM

**Pull request merged**
